pkg/executor/nginx/nginx_test.go

```diff
@@ -776,3 +807,28 @@ func cleanString(in string) string {
 	replacer := strings.NewReplacer("\n", "", "\t", "")
 	return replacer.Replace(in)
 }
+
+func validateNginxConfig(t *testing.T, config string) {
+	//Not relevant for syntax checking
+	config = strings.Replace(config, "include       /etc/nginx/mime.types;", "", -1)
+
+	file, err := ioutil.TempFile("/tmp", "nginx.*.conf")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer os.Remove(file.Name())
+
+	err = ioutil.WriteFile(file.Name(), []byte(config), 0)
+	if err != nil {
+		t.Error(err)
+	}
+
+	cmd := exec.Command("nginx", "-t", "-c", file.Name())
+	result, err := cmd.CombinedOutput()
+
+	if !strings.Contains(string(result), "syntax is ok") && err != nil {
```

Comment:
I suggest:
* If running tests verbosely, log the full output regardless.
* checking the error output to see if the nginx command was not found. In that case, warn-log that it was not found.
* Only fail if nginx was found and the output was not "syntax is ok"

Reply:
Trouble with accepting missing nginx, is that it might pass on Jenkins. I think it is better to make nginx mandatory for local development.